File: frontend/src/types/apiKey.ts
export type ApiKeyScope = 'read' | 'write'

export type ApiKeyStatus = 'active' | 'revoked' | 'expired'

export interface ApiKey {
  id: string
  tenant_id: string
  name: string
  description?: string
  key_prefix: string
  scope: ApiKeyScope
  created_by: string
  expires_at: string | null
  revoked_at: string | null
  last_used_at: string | null
  created_at: string
}

export interface ApiKeyCreateInput {
  tenant_id: string
  name: string
  description?: string
  scope: ApiKeyScope
  expires_at: string | null
}

export interface ApiKeyCreated extends ApiKey {
  plaintext: string
}

export interface ApiKeyListParams {
  tenant_id?: string
  scope?: ApiKeyScope
  status?: ApiKeyStatus
  q?: string
}

export function deriveKeyStatus(k: ApiKey): ApiKeyStatus {
  if (k.revoked_at) return 'revoked'
  if (k.expires_at && new Date(k.expires_at) < new Date()) return 'expired'
  return 'active'
}

File: backend/app/schemas/api_key.py
"""API Key schemas."""
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field

from app.schemas.common import ORMBase


class ApiKeyCreate(BaseModel):
    tenant_id: int
    name: str = Field(min_length=1, max_length=128)
    description: Optional[str] = Field(default=None, max_length=512)
    scope: str = Field(default="read", pattern=r"^(read|write)$")
    expires_at: Optional[datetime] = None


class ApiKeyUpdate(BaseModel):
    name: Optional[str] = Field(default=None, max_length=128)
    description: Optional[str] = Field(default=None, max_length=512)


class ApiKeyListParams(BaseModel):
    tenant_id: Optional[int] = None
    scope: Optional[str] = None
    status: Optional[str] = None  # active | expired | revoked
    q: Optional[str] = None


class ApiKeyOut(ORMBase):
    id: int
    tenant_id: int
    name: str
    description: Optional[str] = None
    key_prefix: str
    scope: str
    created_by: Optional[str] = None
    expires_at: Optional[datetime] = None
    revoked_at: Optional[datetime] = None
    last_used_at: Optional[datetime] = None
    created_at: datetime


class ApiKeyCreated(ApiKeyOut):
    # 仅创建/轮换时返回一次明文
    plaintext: str
